src/com/google/cloud/datastore/types.js

```diff
@@ -146,18 +146,16 @@ foam.LIB({
       var id = key.name;
 
       var o = cls.create();
-
-      if ( cls.ids && cls.ids.length > 1 ) {
-        throw new Error('Not implemented: Deserialization of Cloud Datastore ' +
-            'multi-part ids');
+      if ( cls.model_.ids && cls.model_.ids.length === 1 ) {
+        o[cls.model_.ids[i]] = id;
```

Comment:
This line is broken in two parts: [i] should be [0] and it isn't even safe anyway, since you don't know if cls.model_.ids event exists.
 
This line should be just: o.id = id;